Tests/Marketplace/marketplace_services.py

```diff
@@ -2905,10 +2923,41 @@ def _changelog_entry_bc_versions(predecessor_version: Version, rn_version: Versi
         return {str(bc_ver): bc_version_to_text.get(str(bc_ver)) for bc_ver in breaking_changes_versions if
                 predecessor_version < bc_ver <= rn_version}
 
+    def get_version_to_pr_numbers(self, release_notes_dir) -> Dict[str, List[int]]:
+        """
+        Get Dict[Version,List[PullRequests]] for the given directory
+        Args:
+            release_notes_dir: the directory to find the pull request numbers for
+
+        Returns:
+            a dict of version -> List[Pull Request Numbers] for each file in the release_notes_dir
+        """
+
+        def file_path(file):
+            return f"Packs/{self.name}/{self.RELEASE_NOTES}/{file}"
+
+        if not os.path.exists(release_notes_dir):
+            return {}
+        return {underscore_file_name_to_dotted_version(file): get_pull_request_numbers_from_file(file_path(file))
+                for file in filter_dir_files_by_extension(release_notes_dir, '.md')}
```

Comment:
As much as it is more elegant it is not very readable, let's create a regular for loop something like this:
```suggestion
        versions_dict = {}
        for file in filter_dir_files_by_extension(release_notes_dir, '.md'):
            rn_version =  underscore_file_name_to_dotted_version(file)
            pr_numbers = get_pull_request_numbers_from_file(file_path(file))
            versions_dict[rn_version] = pr_numbers

            return versions_dict
```